LaTeX/Inhalt/04_Inhalt/domain_driven_design.tex

```diff
@@ -0,0 +1,50 @@
+
+\section{Entities}
+Ein \href{https://github.com/moorts/Morik/blob/main/src/application/DDD/Entry.h}{\textit{Entry}} beschreibt einen konreten Eintrag in der Datenbank. Der Entry beinhaltet die ID des Eintrags, den Namen des Eintrags, einen optionalen Login und das verschlüsselte Passwort. Man stelle sich Folgendes vor. Der Benutzer löscht einen Entry und erstellt danach einen neuen. Bei dem neuen Entry vergibt er den gleichen Namen, Login und das gleiche Passwort wie bei dem zuvor gelöschten Entry. Handelt es sich darum um den selben Entry? Nein, denn der wurde gelöscht. Darum ist der Entry eine Entity und kein Value Object.
+
+\section{Aggregate}
+Es gibt genau ein Aggregat. Dieses Aggregat beinhaltet ein Entry, eine EntryId, einen EntryName, einen Login und ein EncryptedPassword. Diese Teile sind als Aggregat zu betrachten, da sie immer gemeinsam von Interesse sind. Ein Benutzer möchte nicht einfach irgend ein Passwort wissen ohne den zugehörigen Namen des Eintrags zu kennen. Die Aggregat-Root ist dabei der Entry, wobei über die EntryId des Entrys auf das jeweilige Aggregat zugegriffen wird.
+
+\section{Repositories}
+Da es nur ein Aggregat gibt, gibt es ebenfalls genau eine Repository. Diese Repository ist die \href{https://github.com/moorts/Morik/blob/main/src/application/DDD/EntryRepository.h}{\textit{EntryRepository}} und stellt den Zugriff auf die Datenbank dar, die die Einträge des Benutzers persistiert. Die EntryRepository liefert demnach vorhandene Entries zurück, speichert neue Entries, löscht nicht mehr benötigte und passt Werte eines Entrys an neue Werte an.
```

Comment:
Ich würde "das Repository" schreiben (zu deutsch "das Depot"), aber not sure was besser ist